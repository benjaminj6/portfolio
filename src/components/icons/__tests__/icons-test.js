/* eslint-env jest */
import { h } from 'preact'
import { shallow } from 'preact-render-spy'

import * as Icons from '../'

test('GitHub Logo snapshot', () => {
  const tree = shallow(<Icons.GithubLogoIcon />)
  expect(tree).toMatchSnapshot()
})

test('LinkedIn Logo snapshot', () => {
  const tree = shallow(<Icons.LinkedInLogoIcon />)
  expect(tree).toMatchSnapshot()
})

test('Medium Logo snapshot', () => {
  const tree = shallow(<Icons.MediumLogoIcon />)
  expect(tree).toMatchSnapshot()
})

describe('Menu Icon', () => {
  test('closed snapshot', () => {
    const tree = shallow(<Icons.MenuIcon open={false} />)
    expect(tree).toMatchSnapshot()
  })

  test('open snapshot', () => {
    const tree = shallow(<Icons.MenuIcon open />)
    expect(tree).toMatchSnapshot()
  })
})

test('Twitter Logo snapshot', () => {
  const tree = shallow(<Icons.TwitterLogoIcon />)
  expect(tree).toMatchSnapshot()
})
