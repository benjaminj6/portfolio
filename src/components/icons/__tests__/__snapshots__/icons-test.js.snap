// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitHub Logo snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<SocialMediaCssWrapper>
  <svg
    width="24px"
    height="24px"
    viewBox="0 0 24 24"
  >
    <path
      fill="#000000"
      d="M12,2A10,10 0 0,0 2,12C2,16.42 4.87,20.17 8.84,21.5C9.34,21.58 9.5,21.27 9.5,21C9.5,20.77 9.5,20.14 9.5,19.31C6.73,19.91 6.14,17.97 6.14,17.97C5.68,16.81 5.03,16.5 5.03,16.5C4.12,15.88 5.1,15.9 5.1,15.9C6.1,15.97 6.63,16.93 6.63,16.93C7.5,18.45 8.97,18 9.54,17.76C9.63,17.11 9.89,16.67 10.17,16.42C7.95,16.17 5.62,15.31 5.62,11.5C5.62,10.39 6,9.5 6.65,8.79C6.55,8.54 6.2,7.5 6.75,6.15C6.75,6.15 7.59,5.88 9.5,7.17C10.29,6.95 11.15,6.84 12,6.84C12.85,6.84 13.71,6.95 14.5,7.17C16.41,5.88 17.25,6.15 17.25,6.15C17.8,7.5 17.45,8.54 17.35,8.79C18,9.5 18.38,10.39 18.38,11.5C18.38,15.32 16.04,16.16 13.81,16.41C14.17,16.72 14.5,17.33 14.5,18.26C14.5,19.6 14.5,20.68 14.5,21C14.5,21.27 14.66,21.59 15.17,21.5C19.14,20.16 22,16.42 22,12A10,10 0 0,0 12,2Z"
    >
    </path>
  </svg>
</SocialMediaCssWrapper>

`;

exports[`LinkedIn Logo snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<SocialMediaCssWrapper>
  <svg
    width="24px"
    height="24px"
    viewBox="0 0 24 24"
  >
    <path
      fill="#000000"
      d="M21,21H17V14.25C17,13.19 15.81,12.31 14.75,12.31C13.69,12.31 13,13.19 13,14.25V21H9V9H13V11C13.66,9.93 15.36,9.24 16.5,9.24C19,9.24 21,11.28 21,13.75V21M7,21H3V9H7V21M5,3A2,2 0 0,1 7,5A2,2 0 0,1 5,7A2,2 0 0,1 3,5A2,2 0 0,1 5,3Z"
    >
    </path>
  </svg>
</SocialMediaCssWrapper>

`;

exports[`Medium Logo snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<SocialMediaCssWrapper>
  <svg
    width="24px"
    height="24px"
    viewBox="0 0 24 24"
  >
    <path
      fill="#000000"
      d="M4.37,7.3C4.4,7.05 4.3,6.81 4.12,6.65L2.25,4.4V4.06H8.05L12.53,13.89L16.47,4.06H22V4.4L20.4,5.93C20.27,6.03 20.2,6.21 20.23,6.38V17.62C20.2,17.79 20.27,17.97 20.4,18.07L21.96,19.6V19.94H14.12V19.6L15.73,18.03C15.89,17.88 15.89,17.83 15.89,17.59V8.5L11.4,19.9H10.8L5.57,8.5V16.14C5.5,16.46 5.63,16.78 5.86,17L7.96,19.57V19.9H2V19.57L4.1,17C4.33,16.78 4.43,16.46 4.37,16.14V7.3Z"
    >
    </path>
  </svg>
</SocialMediaCssWrapper>

`;

exports[`Menu Icon closed snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<div
  open={false}
  class="jsx-3501131796 jsx-3501131796 MenuIcon"
>
  <div class="jsx-3501131796 menuBar"></div>
  <div class="jsx-3501131796 menuBar"></div>
  <div class="jsx-3501131796 menuBar"></div>
  <JSXStyle
    styleId="3501131796"
    css=".jsx-3501131796{--color:var(--black);--transition:300ms ease-in-out;}.MenuIcon.jsx-3501131796{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.menuBar.jsx-3501131796{height:2px;width:100%;background:var(--color);-webkit-transform-origin:0 0;-ms-transform-origin:0 0;transform-origin:0 0;-webkit-transition:-webkit-transform var(--transition),opacity var(--transition), background var(--transition);-webkit-transition:transform var(--transition),opacity var(--transition), background var(--transition);transition:transform var(--transition),opacity var(--transition), background var(--transition);}.menuBar.jsx-3501131796+.menuBar.jsx-3501131796{margin-top:3px;}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796{--color:var(--white);}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796:nth-child(1){-webkit-transform:translate(4px,-1px) rotate(45deg);-ms-transform:translate(4px,-1px) rotate(45deg);transform:translate(4px,-1px) rotate(45deg);}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796:nth-child(2){opacity:0;}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796:nth-child(3){-webkit-transform:translate(2.5px,0.25px) rotate(-45deg);-ms-transform:translate(2.5px,0.25px) rotate(-45deg);transform:translate(2.5px,0.25px) rotate(-45deg);}"
  >
  </JSXStyle>
</div>

`;

exports[`Menu Icon open snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<div
  open={true}
  class="jsx-3501131796 jsx-3501131796 MenuIcon"
>
  <div class="jsx-3501131796 menuBar"></div>
  <div class="jsx-3501131796 menuBar"></div>
  <div class="jsx-3501131796 menuBar"></div>
  <JSXStyle
    styleId="3501131796"
    css=".jsx-3501131796{--color:var(--black);--transition:300ms ease-in-out;}.MenuIcon.jsx-3501131796{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.menuBar.jsx-3501131796{height:2px;width:100%;background:var(--color);-webkit-transform-origin:0 0;-ms-transform-origin:0 0;transform-origin:0 0;-webkit-transition:-webkit-transform var(--transition),opacity var(--transition), background var(--transition);-webkit-transition:transform var(--transition),opacity var(--transition), background var(--transition);transition:transform var(--transition),opacity var(--transition), background var(--transition);}.menuBar.jsx-3501131796+.menuBar.jsx-3501131796{margin-top:3px;}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796{--color:var(--white);}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796:nth-child(1){-webkit-transform:translate(4px,-1px) rotate(45deg);-ms-transform:translate(4px,-1px) rotate(45deg);transform:translate(4px,-1px) rotate(45deg);}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796:nth-child(2){opacity:0;}[open='true'].jsx-3501131796 .menuBar.jsx-3501131796:nth-child(3){-webkit-transform:translate(2.5px,0.25px) rotate(-45deg);-ms-transform:translate(2.5px,0.25px) rotate(-45deg);transform:translate(2.5px,0.25px) rotate(-45deg);}"
  >
  </JSXStyle>
</div>

`;

exports[`Twitter Logo snapshot 1`] = `
preact-render-spy (1 nodes)
-------
<SocialMediaCssWrapper>
  <svg
    width="24px"
    height="24px"
    viewBox="0 0 24 24"
  >
    <path
      fill="#000000"
      d="M22.46,6C21.69,6.35 20.86,6.58 20,6.69C20.88,6.16 21.56,5.32 21.88,4.31C21.05,4.81 20.13,5.16 19.16,5.36C18.37,4.5 17.26,4 16,4C13.65,4 11.73,5.92 11.73,8.29C11.73,8.63 11.77,8.96 11.84,9.27C8.28,9.09 5.11,7.38 3,4.79C2.63,5.42 2.42,6.16 2.42,6.94C2.42,8.43 3.17,9.75 4.33,10.5C3.62,10.5 2.96,10.3 2.38,10C2.38,10 2.38,10 2.38,10.03C2.38,12.11 3.86,13.85 5.82,14.24C5.46,14.34 5.08,14.39 4.69,14.39C4.42,14.39 4.15,14.36 3.89,14.31C4.43,16 6,17.26 7.89,17.29C6.43,18.45 4.58,19.13 2.56,19.13C2.22,19.13 1.88,19.11 1.54,19.07C3.44,20.29 5.7,21 8.12,21C16,21 20.33,14.46 20.33,8.79C20.33,8.6 20.33,8.42 20.32,8.23C21.16,7.63 21.88,6.87 22.46,6Z"
    >
    </path>
  </svg>
</SocialMediaCssWrapper>

`;
